apps/retrospective/serializers/validators/retrospective.go

```diff
@@ -26,7 +27,14 @@ func IsValidTaskProviderConfigList(
 	var isValid bool
 	var providerType string
 	var providerData, credentials map[string]interface{}
-	taskProviderConfigList := currentStruct.Interface().(*retrospectiveSerializers.RetrospectiveCreateSerializer).TaskProviderConfig
+	var taskProviderConfigList []map[string]interface{}
+
+	switch reflect.TypeOf(currentStruct.Interface()) {
+	case reflect.TypeOf(&retrospectiveSerializers.RetrospectiveCreateSerializer{}):
+		taskProviderConfigList = currentStruct.Interface().(*retrospectiveSerializers.RetrospectiveCreateSerializer).TaskProviderConfig
```

Comment:
This statement can be used with the above statement, to make it look a bit better, also we can club these two types in a single case.
```suggestion
		taskProviderConfigList = interfaceValue.TaskProviderConfig
```